networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeMessageHandler.java

```diff
@@ -244,7 +247,18 @@ SafeFuture<Optional<BlobSidecar>> loadNextBlobSidecar() {
             .thenCompose(
                 block -> {
                   maybeCurrentBlock = block;
-                  return retrieveBlobSidecar();
+                  final MiscHelpersDeneb miscHelpersDeneb =
+                      spec.atSlot(currentSlot.get()).miscHelpers().toVersionDeneb().orElseThrow();
+                  blobSidecarsCount = miscHelpersDeneb.getBlobSidecarsCount(block);
+                  if (blobSidecarsCount > maxBlobsPerBlock.intValue()) {
```

Comment:
I'm not sure we need this check. We are getting it from our database, we made all sanity checks before saving it

Reply:
Right, removed it